AxisRotationConstraint.cs

```diff
@@ -0,0 +1,138 @@
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.SpatialManipulation;
+using UnityEngine;
+
+public class AxisRotationConstraint : TransformConstraint
+{
+    #region Properties
+
+    [SerializeField]
+    [Tooltip("Set to true, if rotation around X-Axis is possible.")]
+    private bool allowRotationOnXAxis;
+
+    [SerializeField]
+    [Tooltip("Set to true, if rotation around Y-Axis is possible.")]
+    private bool allowRotationOnYAxis;
+
+    [SerializeField]
+    [Tooltip("Set to true, if rotation around Z-Axis is possible.")]
+    private bool allowRotationOnZAxis;
+
+    [SerializeField]
+    [Tooltip("Set the size of the steps for fixed rotation")]
+    private float fixedStepSize = 1.0f;
+
+    [SerializeField]
+    [Tooltip("Enable to allow rotation only in fixed steps.")]
+    private bool bFixedStep = false;
+
+    [SerializeField]
+    [Tooltip("Check if object rotation should be in local space of object being manipulated instead of world space.")]
+    private bool useLocalSpaceForConstraint = false;
+
+    /// <summary>
+    /// Gets or sets whether the constraints should be applied in local space of the object being manipulated or world space.
+    /// </summary>
+    public bool UseLocalSpaceForConstraint
+    {
+        get => useLocalSpaceForConstraint;
+        set => useLocalSpaceForConstraint = value;
+    }
+
+    public override TransformFlags ConstraintType => TransformFlags.Rotate;
+
+    #endregion Properties
+
+
+    private void Awake()
+    {
+        allowRotationOnXAxis = true;
+        allowRotationOnYAxis = true;
+        allowRotationOnZAxis = true;
+    }
+
+
+    #region Public Methods
+
+    /// <summary>
+    /// Removes rotation about given axis if its flag is found
+    /// in ConstraintOnRotation
+    /// </summary>
+    public override void ApplyConstraint(ref MixedRealityTransform transform)
+    {
+        Quaternion rotation = transform.Rotation * Quaternion.Inverse(WorldPoseOnManipulationStart.Rotation);
+        Vector3 eulers = rotation.eulerAngles;
+
+        if (bFixedStep)
+        {
+            Vector3 eulerStart = Quaternion.Inverse(WorldPoseOnManipulationStart.Rotation).eulerAngles;
+
+            eulers.x = ((int)(eulers.x / fixedStepSize)) * fixedStepSize;
+            eulers.y = ((int)(eulers.y / fixedStepSize)) * fixedStepSize;
+            eulers.z = ((int)(eulers.z / fixedStepSize)) * fixedStepSize;
+        }
+
+
+        if (!allowRotationOnXAxis)
+            eulers.x = 0;
+        
+        if (!allowRotationOnYAxis)
+            eulers.y = 0;
+        
+        if (!allowRotationOnZAxis)
+            eulers.z = 0;
+        
+
+        transform.Rotation = useLocalSpaceForConstraint
+            ? WorldPoseOnManipulationStart.Rotation * Quaternion.Euler(eulers)
+            : Quaternion.Euler(eulers) * WorldPoseOnManipulationStart.Rotation;
+    }
+
+    /// <summary>
+    /// checks, if user can rotate around given axis or not

```

Comment:
Fix grammar in comment 

May be "This check if the user can rotate around a given axis."

and please mark boolean values with "langword" tags...and fix grammar

```
<returns><see langword="true"/> if rotation is possible, otherwise <see langword="false"/>.</returns>
```

Reply:
This comment applies to all XML comment blocks